### DISCUSSION TITLE
bugs related the mpirun when i am trying to run the optimization

### DISCUSSION THREAD

**Discussion by MachAeroDeveloper** [mdolab/dafoam#584] [Bug]
Dear Professor He,

I am writing to you today to seek your assistance with an issue I'm encountering while using Dafoam for airfoil optimization.

I previously ran a successful simulation for airfoil case single point optimization using Dafoam 2.2.5. However, when I attempted to run another simulation with another flow condition , I encountered the following error:
```

*** The MPI_Comm_get_attr() function was called after MPI_FINALIZE was invoked.
*** This is disallowed by the MPI standard.
*** Your MPI job will now abort.
[a851d53b120d:318] Local abort after MPI_FINALIZE completed successfully; not able to aggregate error messages, and not able to guarantee that all other processes were killed!
```

I have tried the following to troubleshoot the issue:

Checked the syntax of my code for errors.
Searched online for similar issues and potential solutions, but nothing has worked so far.
I am attaching the relevant code snippets and error logs for your reference:

![image](https://github.com/DAFoam/workshops/assets/157682082/052c8429-26e9-43ea-96b5-9aec2d5475e6)
![image](https://github.com/DAFoam/workshops/assets/157682082/13706072-4230-4ca0-8441-7fbaea15a0d9)
Specifically, I'm wondering if you could help me understand the following:

What does the error message mean?
What could be causing this issue in my modified case?
Are there any specific steps I can take to try to resolve it?
I am available to provide further information or clarification if needed.

Thank you for your time and assistance.

Sincerely,
Ajlan
[logMeshGeneration.txt](https://github.com/mdolab/dafoam/files/14232528/logMeshGeneration.txt)
[logout.txt](https://github.com/mdolab/dafoam/files/14232529/logout.txt)

**Comment by friedenhe** [comment]
The DAFoam tutorial is for the latest version of DAFoam. The old DAFoam version may not work, please use DAFoam-v3.1.0

**Reply by MachAeroDeveloper** [reply]
thank you professor for quick reply 
i have download the different version of Dafoam of 3.0.0 ,2.2.5 ,3.0.6 ,and this  and this tuturial is from the videos and slide which are version 2.2.5 now when i run on this version v2.2.5 it show me this error 
![image](https://github.com/mdolab/dafoam/assets/157682082/f7eaaa9e-cc76-4b44-9e48-903cf272114e)
[outlog dafoamv2.2.5.txt](https://github.com/mdolab/dafoam/files/14260682/outlog.dafoamv2.2.5.txt)

but when i run the same totorial on the version 3.1.0 i get this error that i have attach them .
![image](https://github.com/mdolab/dafoam/assets/157682082/4d81e531-6419-4e19-80ad-e5718b5b1641)
![image](https://github.com/mdolab/dafoam/assets/157682082/4e9522cd-f764-4528-9b31-f4c57c19a8d7)

please suggest what i should to do what is my mistake i have attach the file of runscript.py 
[out.txt](https://github.com/mdolab/dafoam/files/14261184/out.txt)

**Comment by friedenhe** [comment]
Are you sure you are using the latest DAFoam tutorial? The error said you are using an outdated API for DVGeo in runScript.py

**Reply by MachAeroDeveloper** [reply]
hello professor ,i appreciate your help 
yes i use the last version as you can see on my attachment screen shot of all the version that i have been used 
![Capture](https://github.com/mdolab/dafoam/assets/157682082/70d674fc-f527-46eb-a7a0-8af6c4a0ff85)
![Capture](https://github.com/mdolab/dafoam/assets/157682082/1ceac9e4-a3f0-4366-961a-f38ab98dd43f)
 as you can see here i use the same version and i get this error i do not what i should to do
[logMeshGeneration.txt](https://github.com/mdolab/dafoam/files/14276905/logMeshGeneration.txt)
here is the mesh generation file ,please check it if there is a problem 
![1](https://github.com/mdolab/dafoam/assets/157682082/981fae03-1eca-41d8-8dcb-a9b0c417e2d9)
![2](https://github.com/mdolab/dafoam/assets/157682082/645b7aa7-6ec8-4242-abbd-c18f2f146924)
![3](https://github.com/mdolab/dafoam/assets/157682082/7198da13-feb8-4e62-88d7-a630b40b42bb)

 and when i run the runscript,py on the last version tutorial i get this error 
![6](https://github.com/mdolab/dafoam/assets/157682082/43ac7566-2549-48c8-8f3c-eaeb99ae32b4)

**Comment by friedenhe** [comment]
Are you using runScript.py from one of the DAFoam workshops? If yes, you need to use an old version of DAFoam; check the first workshop slide. If you use runScript.py from https://github.com/dafoam/tutorials, you should use the latest version of DAFoam.

**Reply by MachAeroDeveloper** [reply]
dear professor friedenhe 
thank you so much for your reply , I am writing to seek your guidance regarding an issue I encountered while working on the incompressible NACA0012 airfoil shape optimization  using the latest version of dafoam v3.1.0.

As per your instructions, I have been utilizing the most recent version of the software for our project. However, when I attempt to run the runscript.py file, I encounter an error that I cannot resolve. I have not made any modifications to the script, ensuring it remains in its original state as you provided.

Attached to this is a screenshot of the error message that appears when I run the script. This will provide you with a detailed view of the issue. I believe the screenshot will help in diagnosing the problem more accurately.

Your expertise and advice on how to proceed would be greatly appreciated. I am eager to continue with the project but find myself at a standstill due to this technical hurdle.

Thank you very much for your time and assistance. I look forward to your valuable input.

Best regards,
![image](https://github.com/mdolab/dafoam/assets/157682082/4b7b608e-6852-4930-a4bc-43d9e05b30ad)

![image](https://github.com/mdolab/dafoam/assets/157682082/017773e9-9ad4-44b5-ad5e-939d4d59aff1)
![image](https://github.com/mdolab/dafoam/assets/157682082/713c1ac4-3152-4701-a1c2-31d4fb59ffe2)
![image](https://github.com/mdolab/dafoam/assets/157682082/5e87b6de-ccaa-45ce-9960-c0f810ea72fe)
![image](https://github.com/mdolab/dafoam/assets/157682082/540a9f3a-a55d-41b7-a903-d2c6b6d756ed)

**Comment by friedenhe** [comment]
I just ran the latest tutorial using DAFoam 3.1.0 without a problem. Not sure why your case did not work. Did you manage to follow the "Get Started" section to run your first case?
